pkg/repository/manifest/repo.go

```diff
@@ -57,7 +78,75 @@ func Init(dst string, initTime time.Time) error {
 		return fmt.Errorf("manifest '%s' not initialized porperly", ty)
 	}
 
-	return BatchSaveManifests(dst, manifests)
+	return BatchSaveManifests(dst, manifests, pubKey, privKey)
+}
+
+// AddComponent adds a new component to an existing repository
+func AddComponent(id, name, desc, owner, repo string, isDefault bool, pub, priv string) error {
+	// read key files
+	privBytes, err := ioutil.ReadFile(priv)
+	if err != nil {
+		return err
+	}
+	privKey := &crypto.RSAPrivKey{}
+	if err = privKey.Deserialize(privBytes); err != nil {
+		return err
+	}
+	pubBytes, err := ioutil.ReadFile(pub)
+	if err != nil {
+		return err
+	}
+	pubKey := &crypto.RSAPubKey{}
+	if err = pubKey.Deserialize(pubBytes); err != nil {
+		return err
+	}
+
+	// read manifest files from disk
+	manifests, err := ReadManifestDir(repo)
```

Comment:
I think you should only read the index manifest here. Note that we don't save index.json, only n.index.json and you'll need to use the correct version from snapshot.json